config-model/src/main/java/com/yahoo/vespa/model/application/validation/StreamingValidator.java

```diff
@@ -74,6 +75,11 @@ private static void warnStreamingAttribute(String cluster, String schema, Immuta
                 }
             }
             return;
+        } else if (sd.getDataType() == DataType.PREDICATE) {
+            logger.logApplicationPackage(Level.WARNING,
+                    "For search cluster '" + cluster + "', streaming schema '" + schema +
```

Comment:
Consider adding a helper function for creating the warning message that is used here, in `warnStreamingGramMatching()` and `warnStreamingIndexField()`.